src/components/TodoList.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { TodoItem } from './TodoItem';
+
+export function TodoList(props) {
+  const {
+    filter,
+    todos,
+    handleToggleTodo,
+    handleRemoveTodo,
+    handleEditTodo,
+    handleToggleAll,
+  } = props;
+
+  let list = todos;
+
+  if (filter === 'completed') {
+    list = todos.filter(({ completed }) => !completed);
+  } else if (filter === 'active') {
+    list = todos.filter(({ completed }) => completed);
```

Comment:
will be great to calculate it outside of the render

or [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)